### PR TITLE
Add pipeline limitations endpoint

### PR DESCRIPTION
**Pull request by ivadym**
### Added
- Pipeline limitations endpoint

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_cg -t cg -b [THIS-BRANCH-NAME] -a
    ```

### How to test

- [x] `GET applications/RNAPOAR100/pipeline_limitations`
```
[
  {
    "application_id": 285,
    "comment": "",
    "created_at": "Fri, 20 Oct 2023 10:09:06 GMT",
    "id": 1,
    "limitations": "Only transcribed gene fusions will be detected and their successful detection will depend on several factors, such as level of expression of the fusion transcript, percentage of cells harboring the fusion gene, sequencing depth, as well as, quality and input amount of the RNA sample. \r\n\r\nDetection of exon-skipping events within the same gene are outside the scope of the validated analysis. \r\n\r\nOnly the RNAPOAR100 apptag is currently validated using 300ng of input material.",
    "pipeline": "rnafusion",
    "updated_at": null
  },
  {
    "application_id": 285,
    "comment": "",
    "created_at": "Fri, 20 Oct 2023 10:09:44 GMT",
    "id": 2,
    "limitations": "This is a MIP-RNA limitation :)",
    "pipeline": "mip-rna",
    "updated_at": null
  },
  {
    "application_id": 285,
    "comment": "Created/VI",
    "created_at": "Tue, 24 Oct 2023 09:16:33 GMT",
    "id": 3,
    "limitations": "This is a Balsamic limitation :)",
    "pipeline": "balsamic",
    "updated_at": null
  }
]
```

- [x]  `GET applications/WGSPCFC030/pipeline_limitations` (empty application limitations)
```
[]
```

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

File: cg/server/api.py [hunk 1]
```diff
@@ -450,6 +451,18 @@ def parse_application(tag):
     return jsonify(**application.to_dict())
 
 
+@BLUEPRINT.route("/applications/<tag>/pipeline_limitations")
+@is_public
+def get_application_pipeline_limitations(tag):
```
Comment (henrikstranneheim) [inline cg/server/api.py:456]: Add type hint

File: cg/server/api.py [hunk 2]
```diff
@@ -450,6 +451,18 @@ def parse_application(tag):
...
+    """Return application pipeline specific limitations."""
+    application_limitations: list[ApplicationLimitations] = db.get_application_limitations_by_tag(
+        tag
+    )
+    if application_limitations is None:
```
Comment (henrikstranneheim) [inline cg/server/api.py:461]:
```suggestion
    if not application_limitations:
```

File: cg/store/api/find_business_data.py [hunk 3]
```diff
@@ -74,10 +74,18 @@ def get_application_by_case(self, case_id: str) -> Application:
             .sample.application_version.application
         )
 
+    def get_application_limitations_by_tag(self, tag: str) -> list[ApplicationLimitations] | None:
```
Comment (henrikstranneheim) [inline cg/store/api/find_business_data.py:77]: Perhaps out of scope, but this function should be in a read handler

**Review by henrikstranneheim — APPROVED** [review]
Nice 👍

File: cg/store/api/find_business_data.py (hunk 3 above)
Comment (ivadym) [reply cg/store/api/find_business_data.py:77]: Yes, I agree, as some other functions in this class. Maybe we can address it in a future refactoring PR 🤔

**Comment by ivadym** [comment]
Deployed !

**Pull request merged**